core/src/avm2/globals/flash/utils.rs

```diff
@@ -140,6 +140,56 @@ pub fn escape_multi_byte<'gc>(
     Ok(v.into())
 }
 
+fn handle_percent<I>(it: &mut I) -> Option<u8>
```

Comment:
`handle_percent` can be defined as a closure inside `unescape_multi_byte`, so that the `it` parameter and generic won't be needed.

Reply:
The closure borrows the iterator mutably. This clashes with the main loop also using it mutably.